File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/statistikk/enhetsregisteret/Underenhet.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.statistikk.enhetsregisteret;

import lombok.Builder;
import lombok.Data;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles.Orgnr;

@Builder
@Data
public class Underenhet implements Virksomhet {
    private final Orgnr orgnr;
    private final Orgnr overordnetEnhetOrgnr;
    private final String navn;
    private final Næringskode5Siffer næringskode;
    private final int antallAnsatte;
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/fellesdomene/Underenhet.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.fellesdomene

sealed class Underenhet {
    abstract val orgnr: Orgnr

    data class Næringsdrivende(
        override val orgnr: Orgnr,
        val overordnetEnhetOrgnr: Orgnr,
        override val navn: String,
        override val næringskode: Næringskode,
        val antallAnsatte: Int
    ) : Underenhet(), Virksomhet

    data class IkkeNæringsdrivende(override val orgnr: Orgnr) : Underenhet()
}